### ISSUE TITLE
Should we freeze parameters in generator / discriminator when training the other?

### ISSUE THREAD

**Issue by Yuzz1020** [Yangyangii/GAN-Tutorial#1]
Hi, thank you for this great repo, it helps me a lot.

I have a simple question about the training process. Should we freeze parameters in generator / discriminator when training the other? 

It seems when running G_loss.backward(), parameters in discriminator will change with it?

**Comment by Yangyangii** [comment]
Hi, @noNameForMe1994. :)

These codes are not optimized.
The optimizers for G and D are separate and the models call zero_grad() before every update.
Thus, params of D will be not changed.
And you should be careful to freeze D to get the gradient of G.
But, you can freeze G when you calculate fake outputs for D.

Thank you for sharing :)

**Comment by Yuzz1020** [comment]
> Hi, @noNameForMe1994. :)
> 
> These codes are not optimized.
> The optimizers for G and D are separate and the models call zero_grad() before every update.
> Thus, params of D will be not changed.
> And you should be careful to freeze D to get the gradient of G.
> But, you can freeze G when you calculate fake outputs for D.
> 
> Thank you for sharing :)

Thank you for the detailed explain :)
